clang/lib/DPCT/Diagnostics.inc

```diff
@@ -225,3 +225,5 @@ DEF_WARNING(MODULE_LOAD_DATA, 1104, "'%0' should point to a dynamic library load
 DEF_COMMENT(MODULE_LOAD_DATA, 1104, "'{0}' should point to a dynamic library loaded in memory. The dynamic library should supply \"wrapped\" kernel functions.")
 DEF_WARNING(MODULE_LOAD_DATA_EX, 1105, "'%0' should point to a dynamic library loaded in memory. The dynamic library should supply \"wrapped\" kernel functions. Compilation options passed to cuModuleLoadDataEx are not migrated.")
 DEF_COMMENT(MODULE_LOAD_DATA_EX, 1105, "'{0}' should point to a dynamic library loaded in memory. The dynamic library should supply \"wrapped\" kernel functions. Compilation options passed to cuModuleLoadDataEx are not migrated.")
+DEF_WARNING(SIZEOF_FLOAT3, 1106, "The size of sycl::float3 is 16 bytes while in original code size of float3 is 12 bytes. You may need to adjust the code.")
```

Comment:
Are those sizes an implementation detail or came from some standard?

Reply:
for sizeof(sycl::float3), ref to sycl spec:  https://registry.khronos.org/SYCL/specs/sycl-2020/html/sycl-2020.html#memory-layout-and-alignment